src/PersonalAccessTokenFactory.php

```diff
@@ -78,13 +91,18 @@ public function make($userId, string $name, array $scopes = [])
      *
      * @param  mixed  $userId
      * @param  string[]  $scopes
+     * @param  string|null  $provider
      * @return \Psr\Http\Message\ServerRequestInterface
      */
-    protected function createRequest($userId, array $scopes)
+    protected function createRequest($userId, array $scopes, ?string $provider)
     {
-        $config = config('passport.personal_access_client');
+        $config = $provider
+            ? config("passport.personal_access_client.$provider", config('passport.personal_access_client'))
+            : config('passport.personal_access_client');
+
+        $client = isset($config['id']) ? $this->clients->findActive($config['id']) : null;
 
-        if (! $config) {
+        if (! $client || ($client->provider && $client->provider !== $provider)) {
```

Comment:
To be the same as the check on `TokenGuard`:

https://github.com/laravel/passport/blob/8ea1dd41745c89769fd8c4b207c4739eea707e95/src/Guards/TokenGuard.php#L186-L188